viral/ensemble_reactivation.py

```
@@ -790,27 +888,167 @@ def main() -> None:
+    result = []
+
+    for idx in range(data.shape[0]):
+        onset_times = threshold_detect_continuous(
+            data[idx, :],
+            baseline[idx, :],
+        )
+        assert len(onset_times) > 0, "No events found"
+
+        component_response = []
+
+        for i, onset in enumerate(onset_times):
```

Ruff (0.13.3)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)